indico_toolkit/indico_wrapper/dataset.py

```diff
@@ -11,44 +14,76 @@
     AddFiles,
     ProcessFiles,
     DeleteDataset,
+    CreateEmptyDataset,
 )
-from indico_toolkit.indico_wrapper import IndicoWrapper
-
-
-def dataset_id_required(fn):
-    """
-    'Datasets' optionally requires an ID on instantiation. For methods that require that it exists, 
-    throw a runtimeError if 'dataset_id' not set.
-    """
 
-    @wraps(fn)
-    def check_dataset_id(self, *args, **kwargs):
-        if not isinstance(self.dataset_id, int):
-            raise RuntimeError(
-                "You must set the 'dataset_id' attribute to use this method"
-            )
-        return fn(self, *args, **kwargs)
-
-    return check_dataset_id
+from indico_toolkit.indico_wrapper import IndicoWrapper
+from indico_toolkit.pipelines import FileProcessing
 
 
 class Datasets(IndicoWrapper):
-    def __init__(self, client: IndicoClient, dataset_id: int = None):
+    def __init__(self, client: IndicoClient):
```

Comment:
Curious about the change from obj that contains id to basically a collection of functions without state.  Is that for consistency with the other parts of the toolkit?

Reply:
Previously, some functions required the dataset_id arg and other functions required that you to have instantiated the class with the dataset_id.(no clear distinctions as to why/why not in each case)- I chose to remove it from __init__ because a lot of the functionality involves new datasets (no ID initially) or combinations of methods that you might want to call with an id you didn't have at instantiation . To me, it made more sense to ask for it in the methods for consistency (rather than checking if it is passed as an optional arg, then checking if it is already passed at instantiation, occasionally overwriting it if a new dataset is created, etc.).  Also- it is more consistent with other parts of toolkit (that is how Workflows works)